src/middleware/verifySignUp.js

```diff
@@ -0,0 +1,43 @@
+const db = require("../models");
+const User = db.user;
+
+checkDuplicateUsernameOrEmail = (req, res, next) => {
+  // Username
+  User.findOne({
+    where: {
+      username: req.body.username
+    }
+  }).then(user => {
+    if (user) {
+      res.status(400).send({
+        message: "Failed! Username is already in use!"
+      });
+      return;
+    }
+
+    // Email
+    User.findOne({
+      where: {
+        email: req.body.email
+      }
+    }).then(user => {
+      if (user) {
+        res.status(400).send({
+          message: "Failed! Email is already in use!"
+        });
+        return;
+      }
+
+      next();
+    });
+  });
+};
+
+  next();
+};
+
+const verifySignUp = {
+  checkDuplicateUsernameOrEmail: checkDuplicateUsernameOrEmail
+};
```

Comment:
I think that if the key is equal to the value you can do
```
const verifySignUp = {
   checkDuplicateUsernameOrEmail
};
```

Reply:
you're right! I'll change it, thanks!